.github/workflows/node.yml

```diff
@@ -106,7 +142,195 @@ jobs:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download leap binary
+        uses: AntelopeIO/asset-artifact-download-action@v3
+        with:
+          owner: AntelopeIO
+          repo: leap
+          target: '${{needs.versions.outputs.leap-target}}'
+          prereleases: ${{fromJSON(needs.versions.outputs.leap-prerelease)}}
+          file: 'leap.*${{matrix.platform}}.*(x86_64|amd64).deb'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Leap
+        run: |
+          apt-get install -y ./leap*.deb
+
+      - name: Link Leap TestHarness Module (fix for v4 of leap)
+        run: ln -s /usr/share/leap_testing/tests/TestHarness /usr/lib/python3/dist-packages/TestHarness
+
+      - name: Download EOS EVM Contract
+        uses: AntelopeIO/asset-artifact-download-action@v3
+        with:
+          owner: eosnetworkfoundation
+          repo: eos-evm
+          target: '${{needs.versions.outputs.eos-evm-contract-target}}'
+          prereleases: ${{fromJSON(needs.versions.outputs.eos-evm-contract-prerelease)}}
+          file: 'contract.tar.gz'
+          artifact-name: contract.test-actions-off.tar.gz
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract EOS EVM Contract
+        id: evm-contract
+        run: |
+          mkdir contract
+          mv contract.tar.gz contract/
+          cd contract
+          tar xvf contract.tar.gz
+          cd build
+          echo "EVM_CONTRACT=$(pwd)" >> "$GITHUB_OUTPUT"
+
+      - name: Authenticate
+        id: auth
+        uses: AntelopeIO/github-app-token-action@v1
+        with:
+          app_id: ${{ secrets.TRUSTEVM_CI_APP_ID }}
+          private_key: ${{ secrets.TRUSTEVM_CI_APP_KEY }}
+
+      - name: Download EOS EVM Node builddir
+        uses: actions/download-artifact@v3
+        with:
+          name: build.tar.gz
+
+      - name: Extract EOS EVM Node builddir
+        id: evm-node-build
+        run: |
+          mkdir eos-evm-node
+          mv build.tar.gz eos-evm-node/
+          pushd eos-evm-node
+          tar xvf build.tar.gz
+          pushd build
+          echo "EVM_NODE_BUILD=$(pwd)" >> "$GITHUB_OUTPUT"
+          popd
+
+      - name: Install Test Depedencies
+        run: |
+          pip install --upgrade web3
+          apt install -y nodejs
+          apt install -y npm
+          npm install eosjs
+          npm install ethereumjs-util
+          npm install node-fetch
+          npm install http-jsonrpc-server
+          npm install dotenv
+          npm install is-valid-hostname
+
+      - name: Test Leap Integration
+        run: |
+          mkdir test_run_root
+          cd test_run_root
+          ${{ steps.evm-node-build.outputs.EVM_NODE_BUILD }}/tests/nodeos_eos_evm_test.py -v --eos-evm-contract-root ${{ steps.evm-contract.outputs.EVM_CONTRACT }} --eos-evm-build-root ${{ steps.evm-node-build.outputs.EVM_NODE_BUILD }} --use-tx-wrapper ${{ steps.evm-node-build.outputs.EVM_NODE_BUILD }}/peripherals/tx_wrapper
```

Comment:
Should we consider setting up ctest?

Reply:
I'm not sure what that would look like since this is an integration test that requires access to `leap`, `leap-dev`, `cdt`, `eos-evm-contract` which may not be available locally for `ctest`.